Makefile

```diff
@@ -15,26 +15,73 @@ ROOT := $(shell pwd)
 
 all: local-build
 
+GO_VERSION := 1.15
 SCRIPT_PATH := $(ROOT)/scripts/:${PATH}
 SOURCES := $(shell find . -name '*.go')
-LOCAL_BINARY := bin/local-container-endpoints
-LINUX_BINARY := bin/linux-amd64/local-container-endpoints
+BINARY_NAME := local-container-endpoints
+IMAGE_REPO_NAME := amazon/amazon-ecs-local-container-endpoints
+LOCAL_BINARY := bin/local/${BINARY_NAME}
+
+# AMD_DIR and ARM_DIR correspond to arch suffix tags in the codebuild project
+AMD_DIR := amd64
+ARM_DIR := arm64
+
+AMD_BINARY := bin/${AMD_DIR}/${BINARY_NAME}
+ARM_BINARY := bin/${ARM_DIR}/${BINARY_NAME}
 VERSION := $(shell cat VERSION)
 AGENT_VERSION_COMPATIBILITY := $(shell cat AGENT_VERSION_COMPATIBILITY)
 TAG := $(VERSION)-agent$(AGENT_VERSION_COMPATIBILITY)-compatible
 
+.PHONY: generate
+generate: $(SOURCES)
+	PATH=$(SCRIPT_PATH) go generate ./...
+
 .PHONY: local-build
 local-build: $(LOCAL_BINARY)
 
+.PHONY: build-local-image
+build-local-image:
+	docker run -v $(shell pwd):/usr/src/app/src/github.com/awslabs/amazon-ecs-local-container-endpoints \
+		--workdir=/usr/src/app/src/github.com/awslabs/amazon-ecs-local-container-endpoints \
+		--env GOPATH=/usr/src/app \
+		--env ECS_RELEASE=cleanbuild \
+		golang:$(GO_VERSION) make ${LOCAL_BINARY}
+	docker build --build-arg ARCH_DIR=local -t $(IMAGE_REPO_NAME):latest-local .
+
+# build binaries for each architecture into their own subdirectories
 $(LOCAL_BINARY): $(SOURCES)
 	PATH=${PATH} golint ./local-container-endpoints/...
-	./scripts/build_binary.sh ./bin/
+	./scripts/build_binary.sh ./bin/local
 	@echo "Built local-container-endpoints"
 
-.PHONY: generate
-generate: $(SOURCES)
-	PATH=$(SCRIPT_PATH) go generate ./...
+$(AMD_BINARY): $(SOURCES)
+	@mkdir -p ./bin/$(AMD_DIR)
+	TARGET_GOOS=linux GOARCH=amd64 ./scripts/build_binary.sh ./bin/$(AMD_DIR)
+	@echo "Built local-container-endpoints for linux-amd64"
 
+$(ARM_BINARY): $(SOURCES)
+	@mkdir -p ./bin/$(ARM_DIR)
+	TARGET_GOOS=linux GOARCH=arm64 ./scripts/build_binary.sh ./bin/$(ARM_DIR)
+	@echo "Built local-container-endpoints for linux-arm64"
+
+# Relies on ARCH_SUFFIX environment variable which is set in the build
+# environment (e.g. CodeBuild project). Value will either be amd64 or arm64.
+.PHONY: build-image
+build-image:
+	docker run -v $(shell pwd):/usr/src/app/src/github.com/awslabs/amazon-ecs-local-container-endpoints \
+		--workdir=/usr/src/app/src/github.com/awslabs/amazon-ecs-local-container-endpoints \
+		--env GOPATH=/usr/src/app \
+		--env ECS_RELEASE=cleanbuild \
+		golang:$(GO_VERSION) make bin/${ARCH_SUFFIX}/${BINARY_NAME}
```

Comment:
does GOARCH need to be set here?

Reply:
No, I believe we just need to set it in the build_binary script. See this [commit](https://github.com/awslabs/amazon-ecs-local-container-endpoints/pull/59/commits/8ea5381a847c699dedeb135c1a2b3117bbbf3c90#diff-c4914ef202c27bbd0af10a36fb35a813221d15ea12332e61649de07f653966a2R32).